app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java

```diff
@@ -178,15 +166,14 @@ public void onPageSelected(int position) {
                         tabLayout.getTabAt(CONTRIBUTIONS_TAB_POSITION).select();
                         isContributionsFragmentVisible = true;
                         updateMenuItem();
-
                         break;
                     case NEARBY_TAB_POSITION:
                         Timber.d("Nearby tab selected");
                         tabLayout.getTabAt(NEARBY_TAB_POSITION).select();
                         isContributionsFragmentVisible = false;
                         updateMenuItem();
                         // Do all permission and GPS related tasks on tab selected, not on create
-                        ((NearbyFragment)contributionsActivityPagerAdapter.getItem(1)).onTabSelected(onOrientationChanged);
+                        NearbyParentFragmentPresenter.getInstance().onTabSelected();
```

Comment:
Shall we use callbacks for this thing, that way code would be cleaner and by defining contracts even novice contributors would know the intended use case

Reply:
Yes I agree with @ashishkumar468. Let us use callbacks.